core/src/main/java/org/infinispan/commands/CommandsFactory.java

```diff
@@ -226,17 +235,18 @@ public interface CommandsFactory {
 
    /**
     * Builds a ReplaceCommand
-    * @param key key to replace
-    * @param oldValue existing value to check for if conditional, null if unconditional.
-    * @param newValue value to replace with
-    * @param segment the segment of the given key
-    * @param metadata metadata of entry
+    *
+    * @param key         key to replace
+    * @param oldValue    existing value to check for if conditional, null if unconditional.
+    * @param newValue    value to replace with
+    * @param segment     the segment of the given key
+    * @param metadata    metadata of entry
     * @param flagsBitSet Command flags provided by cache
+    * @param returnEntry true if the {@link CacheEntry} is the command response, otherwise returns previous value.
     * @return a ReplaceCommand
     */
    ReplaceCommand buildReplaceCommand(Object key, Object oldValue, Object newValue, int segment, Metadata metadata,
```

Comment:
Sorry if there is confusion here. But we should have the three commands have the similar overloads. Either we have no overload (1 method like this) or have the one overload (2 methods) like put and remove above.